content/post/deeplearning_in_go_part_1.md

```diff
@@ -0,0 +1,393 @@
++++
+title = "Deep Learning from Scratch in Go - Part 1: Equations Are Graphs"
+subtitle = ""
+date = "2017-04-19T08:43:45+10:00"
+latex = true
+draft = true
+
++++
+
+(Author: Chewxy, @chewxy on [Twitter](https://twitter.com/chewxy) and Gophers Slack)
+
+Welcome to the first part of many about writing deep learning algorithms in Go. The goal of this series is to go from having no knowledge at all to implementing a latest paper. 
+
+Deep learning is not new. In fact the idea of deep learning was spawned in the early 1980s. What's changed since then is our computers - they have gotten much much more powerful.  In this blog post we'll start with something familiar, and edge towards building a conceptual model of deep learning. We won't define deep learning for the first few posts, so don't worry so much about the term.
+
+There are a few terms of clarification to be made before we begin proper. In this series, the word "graph" refers to the concept of graph as used in [graph theory](https://en.wikipedia.org/wiki/Graph_(discrete_mathematics)). For the other kind of "graph" which is usually used for data visualization, I'll use the term "chart".
+
+## Computation ##
+I'm going to start by making a claim: all programs can be represented as graphs. This claim is not new, of course. Nor is it bold or revolutionary. It's the fundamental theory that computer scientists have been working on ever since the birth of the field of computation. But you may have missed it. And so to rehash, the logic goes as such:
+
+1. All modern computer programs run on what essentially is a [Turing Machine](https://en.wikipedia.org/wiki/Turing_machine).
+2. All Turing machines are equivalent to untyped lambda calculus (this is commonly known as the [Church-Turing thesis](https://en.wikipedia.org/wiki/Church_Turing_thesis))
+3. Lambda calculus can be represented as graphs.
+4. Therefore all programs can be represented as graphs.
+
+To make this idea more concrete, let's look at a simple program:
+
+```go
+func main() {
+	fmt.Printf("%v", 1+1)	
+}
+``` 
+
+This generates an [abstract syntax tree](https://en.wikipedia.org/wiki/Abstract_syntax_tree) like so:
```

Comment:
A reader might wonder how this tree is generated.  As in, what do I run or what did you run to generate the tree.